protractor.conf.js

```diff
@@ -89,21 +89,28 @@ if (options.saucelabs) {
   }
   // Avoid going over the SauceLabs concurrency limit (5).
   config.maxSessions = 5;
+  config.allScriptsTimeout = 10 * 60 * 1000;
   // List of browsers configurations tested.
   var sauceBrowsers = require('./sauce_browsers.json');
   // Configuration for SauceLabs browsers.
   config.multiCapabilities = sauceBrowsers.map(function(browser) {
     browser['tunnel-identifier'] = options.tunnelIdentifier;
+    browser.maxDuration = 2000;
+    browser.commandTimeout = 600;
+    browser.idleTimeout = 120;
     return browser;
   });
 } else {
-  // Configuration for phantomJS.
+  // Configuration for headless chrome.
+  config.directConnect = true;
   config.seleniumAddress = 'http://localhost:4444/wd/hub';
```

Comment:
I think this can be removed according to doc:
directConnect: true - Your test script communicates directly Chrome Driver or Firefox Driver, bypassing any Selenium Server. If this is true, settings for seleniumAddress and seleniumServerJar will be ignored. If you attempt to use a browser other than Chrome or Firefox an error will be thrown.